try getting shebang limits with fallback to hardcoded 127 characters

finkandreas opened the pull request:
This PR tries to get the shebang limits onlinux platforms. It could fail getting this number for numerous reasons (e.g. linux-headers not installed, the define looks different). But any failure would result in the current behaviour of having 127 characters as hardcoded value.
If it finds the correct value, it will use it.
This would fix :  #43289

haampie commented:
@spackbot fix style

AndresYague commented:
Very useful patch, thanks!

jonasjucker commented:
@finkandreas This error poped-up on Balfrin again.
Coud you give it another try to get this merged?

finkandreas commented:
Is there something that could be done, to get this merged or rejected? I am unsure whether there is something missing to be merged, or if it is considered too unsafe to be merged?

vbrunini commented on lib/spack/spack/hooks/sbang.py:
```diff
@@ -26,6 +27,18 @@
+        # searching for line '#define BINPRM_BUF_SIZE 256' in /usr/include/linux/binfmts.h
+        # the nbr-1 is the sbang limit on the linux platform
+        sbang_limit_re = re.compile("#define BINPRM_BUF_SIZE ([0-9]+)")
+        with open("/usr/include/linux/binfmts.h") as f:
```
```suggestion
        with open("/usr/include/linux/binfmts.h", 'r', encoding='utf-8') as f:
```

tvoskui commented on lib/spack/spack/hooks/sbang.py:
```diff
@@ -26,6 +27,18 @@
+        # searching for line '#define BINPRM_BUF_SIZE 256' in /usr/include/linux/binfmts.h
+        # the nbr-1 is the sbang limit on the linux platform
+        sbang_limit_re = re.compile("#define BINPRM_BUF_SIZE ([0-9]+)")
+        with open("/usr/include/linux/binfmts.h", 'r', encoding='utf-8') as f:
```
```suggestion
        with open("/usr/include/linux/binfmts.h", "r", encoding="utf-8") as f:
```

finkandreas committed:
This PR tries to get the shebang limits onlinux platforms. It could fail
getting this number for numerous reasons (e.g. linux-headers not installed,
the define looks different). But any failure would result in the current
behaviour of having 127 characters as hardcoded value.

If it finds the correct value, it will use it.

Co-authored-by: Victor Brunini <victorbrunini@gmail.com>
Co-authored-by: Todd Gamblin <victorbrunini@gmail.com>
Signed-off-by: finkandreas <finkandreas@web.de>
Signed-off-by: Todd Gamblin <tgamblin@llnl.gov>
